ReactiveCocoa/AppKit/NSPopUpButton.swift

```diff
@@ -0,0 +1,35 @@
+import ReactiveSwift
+import AppKit
+import enum Result.NoError
+
+extension Reactive where Base: NSPopUpButton {
+	
+	/// A signal of selected indexes
+	public var selectedIndexes: Signal<Int, NoError> {
+		return self.integerValues
+			.map { [unowned base = self.base] _ -> Int? in
+				guard let item = base.selectedItem else { return nil }
+				return base.index(of: item)
+			}
+			.skipNil()
+	}
+	
+	/// Sets the button with an index.
+	public var selectedIndex: BindingTarget<Int> {
+		return makeBindingTarget { $0.selectItem(at: $1) }
+	}
+	
+	/// A signal of selected title
+	public var selectedTitles: Signal<String, NoError> {
+		return self.integerValues
+			.map { [unowned base = self.base] _ -> String? in
+				return base.selectedItem?.title
+			}
+			.skipNil()
+	}
+	
+	/// Sets the button with title.
+	public var selectedTitle: BindingTarget<String> {
+		return makeBindingTarget { $0.selectItem(withTitle: $1) }
+	}
```

Comment:
Having attempted at this, NSPopUpButton is always selected and does not have option to allow for "not selected" state. 
This seems to be why `NSPopUpButton.indexOfSelectedItem` is not optional. 
Although `NSPopUpButton.titleOfSelectedItem` is optional, the underlying property `NSMenuItem.title` is not optional.
All of this makes setting and emitting nil values impossible, thus untestable. I don't think these APIs should be optional.